tests/test_detector_date_of_birth.py

```diff
@@ -45,11 +45,12 @@ def test_DoB_5(self):
         """
         self.compare_before_after()
 
-    def test_generate(self):
-        fake = faker.Faker()
-        faker.Faker.seed(4321)
-
-        self.assertEqual(
-            str(datetime.date(1939, 7, 20)),
-            DoBFilth.generate(faker=fake),
-        )
+    # This is not useful if we use contextual filters
```

Comment:
I kinda think you should leave it in since the generate function is defined already. This only tests generate not how it interacts with your detector.